runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -102,6 +104,7 @@ local function tokens_to_ranges(data, bufnr, client, request)
   local token_modifiers = legend.tokenModifiers
   local encoding = client.offset_encoding
   local lines = api.nvim_buf_get_lines(bufnr, 0, -1, false)
+  local line_ending = vim.bo.fileformat[bufnr] == 'dos' and '\r\n' or '\n'
```

Comment:
Seems like we already have an utility function for this: https://github.com/neovim/neovim/blob/5f00e6adafebc7764c64496540bdd703e8a6a24e/runtime/lua/vim/lsp.lua#L81C14-L81C34